src/palimpzest/sets.py

```diff
@@ -193,16 +197,16 @@ def average(self) -> Dataset:
             source=self,
             schema=Number,
             desc="Average results",
-            aggFunc=AggFunc.AVERAGE,
+            agg_func=AggFunc.AVERAGE,
             nocache=self._nocache,
         )
 
-    def groupby(self, groupBy: GroupBySig) -> Dataset:
+    def groupby(self, groupby: GroupBySig) -> Dataset:
         return Dataset(
             source=self,
-            schema=groupBy.outputSchema(),
+            schema=groupby.output_schema(),
             desc="Group By",
-            groupBy=groupBy,
+            groupBy=groupby,
```

Comment:
The key has to be `group_by` based on your change to `Set` (i.e. `group_by=groupby`)

Reply:
(or feel free to change `Set`'s key to `groupby` to match; just note that we do need to keep this function name `groupby()` -- that's the only hard(ish) constraint)

Reply:
Good catch, fixed